posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,553 @@
+        
+        # IF interpolate center density
+        center_ind = self.model_IF.interpolators[0].out_keys.index('S1_log_center_Rho')
+        max_rho = self.model_IF.interpolators[0].test_interpolator(10**inputs)[:,center_ind]                     
+             
+        # reconstruct profile
+        norm_prof = self.pca.inverse_transform(weights_pred*self.scaling)
+        density_profiles = norm_prof*(max_rho[:,np.newaxis]-min_rho[:,np.newaxis]) \
+                               + min_rho[:,np.newaxis]
+        
+        # IF interpolate final mass, construct mass enclosed profile coordinates
+        m1_ind = self.model_IF.interpolators[0].out_keys.index("star_1_mass")
+        pred_mass = self.model_IF.interpolators[0].test_interpolator(10**inputs)[:,m1_ind]
+        mass_coords = np.linspace(0,1,200)*pred_mass[:,np.newaxis] 
+        
+        return mass_coords,density_profiles
+
+                
+class X_H:
+    
+    def __init__(self,initial,profiles,s1,valid_initial, valid_profiles,valid_s1,IF_interpolator):
+        """Creates and trains H mass fraction profile model.
+        Args:
+            initial (array-like) : log-space initial conditions for training data.
+            profiles (array-like) : final H mass fraction profiles for training data.
+            s1 (array-like) : final star 1 state for training data.
+            valid_initial (array-like) : log-space initial conditions for testing data.
+            valid_profiles (array-like) : final H mass fraction profiles for testing data.
+            valid_s1 (array-like) : final star 1 state for testing data.
+            IF_interpolator (string) : path to .pkl file for IF interpolator for central density, final mass values.
+        """
+        self.initial = initial
+        self.profiles = profiles
+        self.s1 = s1
+        self.valid_initial = valid_initial
+        self.valid_profiles = valid_profiles
+        self.valid_s1 = valid_s1
+
+        # load IF interpolator
+        self.model_IF = IFInterpolator()
+        self.model_IF.load(filename=IF_interpolator)
+        self.c_ind = self.model_IF.interpolators[0].out_keys.index("S1_center_h1")
+        self.s_ind = self.model_IF.interpolators[0].out_keys.index("S1_surface_h1")
+        self.interp = self.model_IF.interpolators[0]
+        
+        # star 1 states
+        self.class_names = ['None',
+                             'H-rich_Shell_H_burning',
+                             'H-rich_Central_He_depleted',
+                             'H-rich_non_burning',
+                             'H-rich_Core_H_burning',
+                             'H-rich_Core_He_burning',
+                             'stripped_He_Central_He_depleted',
+                             'stripped_He_non_burning',
+                             'NS',
+                             'stripped_He_Central_C_depletion',
+                             'H-rich_Central_C_depletion',
+                             'BH']
```

Comment:
How do you selected those states? E.g. why is there no "WD"?

Reply:
I just used the ones that showed up in the grid I used to develop this; do you know if others are missing other than "WD"?

Reply:
from plot_defaults.py:
`'H-rich_Core_C_burning'`
`'stripped_He_Core_He_burning'`
`'stripped_He_Core_C_burning'`
There are some more strange states, which should never happen:
`'stripped_He_Core_H_burning'`
`'stripped_He_Shell_H_burning'`
`'undetermined_evolutionary_state'`
By the way, I'd suggest you to bring them in a more logical order.